intelhex/__init__.py

```diff
@@ -630,26 +659,48 @@ def write_hex_file(self, f, write_start_addr=True, eolstyle='native', byte_count
             minaddr = addresses[0]
             maxaddr = addresses[-1]
 
+            if extaddr_rectyp == -1:
```

Comment:
I think we should check maxaddr before checking start addr record type.

Reply:
Reason why this is here, is because whole data is manged here. We have all necessary variables here. If we want to make checks for 'maxaddr' then I'm understanding this that You want to make this more important criteria than Start Address type for Extended Address resolving. If this was Your intention, then no problem - I will do a little change and refactor. If not then I suggest would stay is it is now.